src/themes/types.ts

```diff
@@ -71,6 +71,16 @@ type EmphasisColorsStrict = Partial<{
 
 export type EmphasisColors = Extendable<EmphasisColorsStrict, ColorVariants>
 
+/**
+ * A type for extracting the color names.
+ */
+type ColorNames = keyof (EmphasisColorsStrict & NaturalColorsStrict)
```

Comment:
```suggestion
type ColorNames = keyof (EmphasisColors & NaturalColors)
```

We should use extendable color types, otherwise it will block the usage of custom colors.

Reply:
with your suggestions we don't get the correct keys exactly because `keyof` does not work as expected on extendable types:
![screenshot 2018-12-12 at 19 15 15](https://user-images.githubusercontent.com/5442794/49889618-48f34800-fe42-11e8-8b7c-4987fb815a8b.png)
with my change:
![screenshot 2018-12-12 at 19 15 06](https://user-images.githubusercontent.com/5442794/49889620-4a247500-fe42-11e8-8044-9569ad9f8333.png)

and to address your comment I changed

```tsx
export type ColorValues<T> = Partial<Record<ColorNames, T>>
```
to
```tsx
Extendable<Partial<Record<ColorNames, T>>, T>
```
in order not to block the usage of custom colors. What do you think?

Reply:
LTGM, there is only one thing that confuses me, it's a name of `ColorNames`. It doesn't represent the merge of `EmphasisColors` and `NaturalColors`. May be other folks have a better idea there 💡

Reply:
if anyone has better ideas for name, pls provide some recommendations:

@mnajdova @kuzhelov @miroslavstastny @levithomason